docker/deployment/app-deployment.yaml

```diff
@@ -57,10 +70,19 @@ spec:
       labels:
         app: app
     spec:
+      securityContext: # see https://github.com/sillsdev/web-languageforge/pull/974 for further explanation of the need for a securityContext here
+        fsGroup: 33 # this is www-data from /etc/passwd, k8s requires explicit assignment otherwise the volume will get mounted with root and www-data won't be allowed to create folders on-the-fly in there.
+      volumes:
+      - name: assets
+        persistentVolumeClaim:
+          claimName: lf-project-assets
       containers:
       - name: app
         image: sillsdev/web-languageforge:app-staging
         imagePullPolicy: Always
```

Comment:
Worth considering adding a `securityContext` here with `runAsGroup` matching the `fsGroup` from the pod's security context. That's the default group for the `www-user` account, but at the Kubernetes level where named user & group IDs are not available, only numeric IDs, it's worth making sure that the configuration matches between the volume mount and the container that will be using the volume.

Important to note that the standard UID and GID for the `www-data` user and group are 33 in Debian-based systems, but 82 in Alpine-based systems. So we can't just blithely assume that `www-data`'s GID will always be 33, we need to document it or configure it explicitly. Otherwise if we start using Alpine-based containers mixed with Debian-based containers we could be in for a surprise later on when things don't work.

Reply:
> Important to note that the standard UID and GID for the www-data user and group are 33 in Debian-based systems, but 82 in Alpine-based systems. So we can't just blithely assume that www-data's GID will always be 33, we need to document it or configure it explicitly. Otherwise if we start using Alpine-based containers mixed with Debian-based containers we could be in for a surprise later on when things don't work.

Excellent point, too bad we have to introduce this kind of maintenance, it'll def make these more brittle unfortunately.  I think I'll make a reference in the Dockerfile's where it's most likely to be noticed.

Reply:
> I actually played with those during my testing and it caused problems, the container would not start.

Was there any error output besides "the container would not start"?  It would be helpful to add documentation to this thread if either of you are able to get farther in this problem.  Getting `runAsGroup` working seems like an important component to make this less brittle.  So a question: can we get by with just `runAsGroup` and remove the `fsGRoup` all together?  If so, this would get around the GID-specific issues @rmunn identified between Alpine/Debian

Reply:
yeah, there was a little output but it was ambiguous, something about not starting some IP address or something...it didn't seem helpful to me at the time and I was more focused on simply trying things that would work rather than going down another rabbit trail at the moment.

All of those options still require a gid rather than a name so that brittleness will still be there but I was having some thoughts about another approach where we could chown the name, i.e., www-data, in an init container which would protect us from the gid maintenance.